.github/workflows/minimum_dependency_checker.yml

```diff
@@ -0,0 +1,93 @@
+name: Minimum Dependency Checker
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'requirements.txt'
+      - 'test-requirements.txt'
+      - 'dask-requirements.txt'
+      - 'koalas-requirements.txt'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Run min dep generator - test reqs
+        id: min_dep_gen_test
+        uses: alteryx/minimum-dependency-generator@v2.1
+        with:
+          requirements_paths: 'test-requirements.txt'
+      - name: Save min test deps and run diff
+        id: check_min_test
+        continue-on-error: true
+        run: |
+          mkdir /tmp/dependencies_updated_artifacts
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > /tmp/minimum_test_requirements.txt
+          diff /tmp/minimum_test_requirements.txt woodwork/tests/minimum_test_requirements.txt
+      - name: if min test deps have changed, write output file
+        if: steps.check_min_test.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > woodwork/tests/minimum_test_requirements.txt
```

Comment:
this could be a single line since there is only one command, although I could see using the current approach due to line length

Reply:
To make it easier for a developer reading the workflow, I did the multi-line approach. I tried removing it, and it's much more difficult to read, so I'll keep it as is for now.